src/linter.cc

```diff
@@ -240,6 +243,40 @@ absl::Status CheckAliasKeyword(absl::string_view sql) {
     }).ApplyTo(sql);
 }
 
+absl::Status CheckTabCharactersUniform(absl::string_view sql,
+    const char indent, const char delimeter) {
+    bool is_indent = true;
+    const char kSpace = ' ', kTab = '\t';
+
+    for (int i=0; i<static_cast<int>(sql.size()); ++i) {
+        if ( sql[i] == delimeter ) {
+            is_indent = true;
+        } else if ( sql[i] != indent ) {
+            if ( (sql[i] == kTab || sql[i] == kSpace) && is_indent ) {
+                // Return error on position <i>.
+                ParseLocationPoint lp =
+                    ParseLocationPoint::FromByteOffset(i);
+                ParseLocationTranslator lt(sql);
+                zetasql_base::StatusOr <std::pair <int, int> > status_or_pos =
+                    lt.GetLineAndColumnAfterTabExpansion(lp);
+                ZETASQL_RETURN_IF_ERROR(status_or_pos.status());
+                std::pair <int, int> error_pos = status_or_pos.value();
```

Comment:
There is another convenience macro that allows writing this in 1 line:
`ZETASQL_ASSIGN_OR_RETURN(std::pair<int, int> pos, lt.GetLineAndColumnAfterTabExpansion(lp));`

Reply:
I couldn't figure out why this doesn't work with `std::pair <int, int> error_pos` but works with `auto error_pos` :(
It says
```
src/linter.cc:264:58: error: expected primary-expression before 'int'
  264 |                 ZETASQL_ASSIGN_OR_RETURN(std::pair <int, int> error_pos,
      |                                                          ^~~
```

Reply:
ah, sorry, my bad. This macro doesn't play well with <> characters :( How about:
```
std::pair<int, int> pos;
ZETASQL_ASSIGN_OR_RETURN(pos, lt.GetLineAndColumnAfterTabExpansion(lp));
```

Reply:
This works, thanks!